openverse_catalog/dags/providers/provider_api_scripts/metropolitan_museum.py

```diff
@@ -6,138 +6,167 @@
 )
 logger = logging.getLogger(__name__)
 
-delayed_requester = DelayedRequester(DELAY)
-image_store = ImageStore(provider=PROVIDER)
-
-
-def main(date=None):
-    """
-    This script pulls the data for a given date from the Metropolitan
-    Museum of Art API, and writes it into a .TSV file to be eventually
-    read into our DB.
-
-    Required Arguments:
-
-    date:  Date String in the form YYYY-MM-DD.  This is the date for
-           which running the script will pull data.
-    """
-
-    logger.info(f"Begin: Met Museum API requests for date: {date}")
 
-    fetch_the_object_id = _get_object_ids(date)
-    if fetch_the_object_id:
-        logger.info(f"Total object found {fetch_the_object_id[0]}")
-        _extract_the_data(fetch_the_object_id[1])
-
-    total_images = image_store.commit()
-    logger.info(f"Total CC0 images received {total_images}")
-
-
-def _get_object_ids(date, endpoint=ENDPOINT):
-    query_params = ""
-    if date:
-        query_params = {"metadataDate": date}
-
-    response = _get_response_json(query_params, endpoint)
-
-    if response:
-        total_object_ids = response["total"]
-        object_ids = response["objectIDs"]
-    else:
-        logger.warning("No content available")
-        return None
-    return [total_object_ids, object_ids]
-
-
-def _get_response_json(
-    query_params,
-    endpoint,
-    retries=5,
-):
-    response_json = delayed_requester.get_response_json(
-        endpoint, query_params=query_params, retries=retries
-    )
-
-    return response_json
-
-
-def _extract_the_data(object_ids):
-    for i in object_ids:
-        _get_data_for_image(i)
-
-
-def _get_data_for_image(object_id):
-    object_json = _get_and_validate_object_json(object_id)
-    if not object_json:
-        logger.warning(f"Could not retrieve object_json for object_id: {object_id}")
-        return
-
-    main_image = object_json.get("primaryImage")
-    other_images = object_json.get("additionalImages", [])
-    image_list = [main_image] + other_images
-
-    meta_data = _create_meta_data(object_json)
-
-    for img in image_list:
-        foreign_id = _build_foreign_id(object_id, img)
-        image_store.add_item(
-            foreign_landing_url=object_json.get("objectURL"),
-            image_url=img,
-            license_info=DEFAULT_LICENSE_INFO,
-            foreign_identifier=foreign_id,
-            creator=object_json.get("artistDisplayName"),
-            title=object_json.get("title"),
-            meta_data=meta_data,
+class MetMuseumDataIngester(ProviderDataIngester):
+    providers = {"image": prov.MET_MUSEUM_DEFAULT_PROVIDER}
+    endpoint = "https://collectionapi.metmuseum.org/public/collection/v1/objects"
+    DEFAULT_LICENSE_INFO = get_license_info(license_="cc0", license_version="1.0")
+
+    # adding args for automatically generated parameters from generate_tsv_filenames
+    def __init__(self, date: str = None, *args):
```

Comment:
Started seeing this when testing locally and moved this back to draft. based on [this](https://airflow.apache.org/docs/apache-airflow/stable/faq.html#what-s-the-deal-with-start-date), I think `datetime.datetime` might be better than `str` here, but I'm not sure if a one-of fix is better, or if I could be doing something more coordinated.

```
Broken DAG: [/usr/local/airflow/openverse_catalog/dags/providers/provider_workflow_dag_factory.py] Traceback (most recent call last): File "/usr/local/airflow/openverse_catalog/dags/providers/provider_dag_factory.py", line 165, in create_provider_api_workflow dag = DAG( File "/usr/local/airflow/.local/lib/python3.10/site-packages/airflow/models/dag.py", line 388, in __init__ if start_date and start_date.tzinfo: AttributeError: 'str' object has no attribute 'tzinfo'
```

Reply:
I'm not sure why you're seeing that error, but it looks like that's referring to the `start_date` for the DAG, rather than _this_ date, which is the date for ingestion.

The DAG's `start_date` is [defined in the provider_workflow](https://github.com/WordPress/openverse-catalog/pull/674/files#diff-c9fe199d287d2190c2b675f6056f2bd047894744aa2c891827f8b88a069c602bL135) as a datetime and looks good to me.

The `date` that we get here is generated [here](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/providers/provider_dag_factory.py#L185) as a string (and by default will be today's date). It needs to be a string because it's what we're passing to the provider API in query params.

Reply:
Thank you so much! It was a total typo in the provider workflow. Now I just need to figure out why it's skipping.